packages/yew-macro/src/derive_props/field.rs

```diff
@@ -323,6 +327,34 @@ fn is_path_an_option(path: &Path) -> bool {
     is_path_segments_an_option(path.segments.iter().take(3).map(|ps| ps.ident.to_string()))
 }
 
+fn is_path_segments_a_string(path_segments: impl Iterator<Item = String>) -> bool {
+    fn is_string_path_seg(seg_index: usize, path: &str) -> u8 {
+        match (seg_index, path) {
+            (0, "alloc") => 0b001,
+            (0, "std") => 0b001,
+            (0, "String") => 0b111,
+            (1, "string") => 0b010,
+            (2, "String") => 0b100,
+            _ => 0,
+        }
+    }
+
+    path_segments
+        .enumerate()
+        .fold(0, |flags, (i, ps)| flags | is_string_path_seg(i, &ps))
+        == 0b111
+}
```

Comment:
It feels a bit too much like C shenanigans to me :sweat_smile: 

What about this:

```
let path = path.segments.iter()
    .map(|ps| ps.ident.to_string())
    .collect::<Vec<_>>();

if path == ["std", "string", "String"] {
    return true;
}

if path == ["alloc", "string", "String"] {
    return true;
}

if path == ["String"] {
    return true;
}

// ... (match wont work unfortunately)

return false;
```

Or maybe:

```
let mut it = path.segments.iter()
    .map(|ps| ps.ident.to_string());

return (
    matches!(it.next().as_deref(), Some("std") | Some("alloc")) &&
    it.next().as_deref() == Some("string") &&
    it.next().as_deref() == Some("String")
)
```

Reply:
I copied this approach from a function defined just above this one, `is_path_an_option`, very hacky but also very concise, so I decided to keep the style

Reply:
I would rather make both of them more readable. Bitwise operations can be very hard to understand